extensions/stackdriver/common/context_test.cc

```diff
@@ -0,0 +1,130 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "extensions/stackdriver/common/context.h"
+#include "extensions/stackdriver/common/constants.h"
+#include "google/protobuf/struct.pb.h"
+#include "gtest/gtest.h"
+
+namespace Extensions {
+namespace Stackdriver {
+namespace Common {
+
+// Test ExtraceNodeMetadata with all possible metadata field.
+TEST(ContextTest, ExtractNodeMetadata) {
+  google::protobuf::Struct metadata_struct;
+  auto metadata_fields = metadata_struct.mutable_fields();
+  google::protobuf::Value pod_name;
+  *pod_name.mutable_string_value() = "test_pod";
```

Comment:
You could use TextProto to ease testing here.

Reply:
Done.